File: src/main/java/io/github/takenoha/towerdefense/persistence/VoucherDeliveryResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;

/** Voucher plus its durable delivery operation after a prepare/reconcile read. */
public record VoucherDeliveryResult(
        VoucherDeliveryOutcome outcome,
        ResourceVoucher voucher,
        VoucherDeliveryOperation operation) {
    public VoucherDeliveryResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(voucher, "voucher");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/VoucherDeliveryResult.kt
package io.github.takenoha.towerdefense.persistence

import java.util.Objects
import kotlin.jvm.JvmRecord

/** Voucher plus its durable delivery operation after a prepare/reconcile read. */
@JvmRecord
data class VoucherDeliveryResult(
    val outcome: VoucherDeliveryOutcome,
    val voucher: ResourceVoucher,
    val operation: VoucherDeliveryOperation?,
) {
    init {
        Objects.requireNonNull(outcome, "outcome")
        Objects.requireNonNull(voucher, "voucher")
    }
}
